spec/features/queue_interactions_spec.rb

```diff
@@ -0,0 +1,57 @@
+require 'spec_helper'
+
+feature 'User interacts with my queue' do
+  scenario "aaa" do
+    comedy = Fabricate(:category, name: 'Comedy')
+    red_dwarf = Fabricate(:video, title: 'Red Dwarf', category: comedy)
+    the_office = Fabricate(:video, title: 'The Office', category: comedy)
+    south_park = Fabricate(:video, title: 'South Park', category: comedy)
+
+    sign_in
+
+    add_video_to_queue(red_dwarf)
+    expect_video_to_be_in_queue(red_dwarf)
+
+    visit video_path(red_dwarf)
+    expect_link_not_to_be_seen('+ My Queue')
+
+    add_video_to_queue(the_office)
+    add_video_to_queue(south_park)
+
+    set_video_position(red_dwarf, 3)
+    set_video_position(the_office, 1)
+    set_video_position(south_park, 2)
+
+    press_update_queue_button
+
+    expect_video_position(the_office, 1)
+    expect_video_position(south_park, 2)
+    expect_video_position(red_dwarf, 3)
+  end
+
+  def add_video_to_queue(video)
+    visit home_path
+    find("a[href='/videos/#{video.id}']").click
+    click_link "+ My Queue"
+  end
+
+  def expect_video_to_be_in_queue(video)
+    page.should have_content(video.title)
```

Comment:
I would not use `should` syntax in `expect ...` method or when everywhere else you use `expect`. Keeping same mental model is important long-term.